stdlib/io/src/main/java/org/ballerinalang/stdlib/io/channels/base/readers/AsyncReader.java

```diff
@@ -36,7 +36,7 @@
  */
 public class AsyncReader implements Reader {
 
-    private static final Logger log = LoggerFactory.getLogger(BlockingReader.class);
+    private static final Logger log = LoggerFactory.getLogger(AsyncReader.class);
```

Comment:
Why is this class called AsynReader? Whether the read blocks or not depends on the channel this is been called on right? If the channel is in blocking mode won't the read method will block?